Tests/scripts/collect_tests/test_conf.py

```diff
@@ -44,15 +44,28 @@ def __init__(self, conf_path: Path):
         self.test_id_to_test = {test.playbook_id: test
                                 for test in self.tests}
 
-        self.tests_to_integrations: dict[str, tuple[str, ...]] = {
-            test.playbook_id: test.integrations
-            for test in self.tests
-            if test.integrations
-        }
+        self.tests_to_integrations: dict[str, tuple[str, ...]] = {}
+        for test in self.tests:
+            if integrations := test.integrations:
+                if test.playbook_id not in self.tests_to_integrations:
+                    self.tests_to_integrations[test.playbook_id] = ()
+                self.tests_to_integrations[test.playbook_id] = tuple(
+                    set(
+                        list(self.tests_to_integrations[test.playbook_id]) + list(integrations)
+                    )
+                )
```

Comment:
```suggestion
        tests_to_integration_set: dict[str, set[str, ...]] = defaultdict(set)
        for test in self.tests:
            self.tests_to_integrations[test.playbook_id].update(test.integrations)
        self.tests_to_integrations: dict[str, tuple[str, ...]] = {test: tuple(test_integrations) for test,test_integrations in tests_to_integration_set}
```

Reply:
i think you meant `tests_to_integration_set[test.playbook_id].update(test.integrations)` instead of `self.tests_to_integration_set[test.playbook_id].update(test.integrations)`

Reply:
exactly

Reply:
is there a reason to sort it?

Reply:
output consistency, prevents odd bugs